File: tests/basic/specialize_any.ml
open Optitrust
open Target


let _ = Run.doc_script_cpp (fun _ ->
      !! Specialize_basic.any (expr "i") [cAny];
  )
"
#include \"../../include/optitrust.h\"
int ANY(int);
int main() {
  int i = 0;
  int a = ANY(2);
}
"

let _ = Run.script_cpp (fun _ ->



  !! Specialize_basic.any (lit "2") [sInstr "corners"; cAny];

  !! Specialize_basic.any (var "i") [cAny];
)

File: tests/basic/specialize_any.cpp
int ANY(int maxValue) { return 0; }

int const N = 2;

typedef int T;

void test() {
  int corners[4];
  corners[ANY(4)] = 9;
} 

int main() {
  T a;
  T x[N];
  x[0] = a;
  for (int k = 1; (k < N); k++) {
    x[k] = 0;
  }
  for (int i = 0; (i < N); i++) {
    x[ANY(N)]++;
  }
  return 0;
}